### DISCUSSION TITLE
Want to specify the images to be used as training and as validation set

### DISCUSSION THREAD

**Discussion by ozgurcemozgen06** [talmolab/sleap#1064] [Help!] [answered]
Hi, 

I aim to compare the model evaluation between DeepLabCut and SLEAP. For this purpose, I want to use the same images as training set that were used as training set in evaluating DeepLabCut, i.e., I do not want the model to randomly select images for training, I want to specify those images. I would be so glad if anyone may guide me how to do it 

Best regards

**Comment by roomrys** [comment, marked as answer]
Hi @ozgurcemozgen06,

There are two ways of doing this, either by specifying the `training_inds` and `validation_inds` in the model's training_config.json or by creating two separate SLEAP projects - one containing all the training frames and one containing all the validation frames.

The easiest method to do manually is likely the second method (training/validation indices are stored in the training_config automatically by SLEAP for reproducibility, but it'd be a bit involved to do yourself).

### Use the CLI: [`sleap-train <path/to/training_config.json> <path/to/train/slp> --val_labels <path/to/validation/slp>`](https://sleap.ai/guides/cli.html?highlight=cli#sleap-train)
You will need two separate SLEAP projects. One containing annotations/labels for all the frames to be used for training. And another SLEAP project containing annotations/labels for all the frames to be used for validation. 
>Note that only the user-labeled frames will be used for training/validation.

If you already have a single slp project, then you will need to break the slp into the two slp's specified above (I can provide some code for this if it is the case). Once you have your training slp and your validation slp, you will need to open the terminal. In the `sleap` conda environment, run the command:
```
sleap-train <path/to/training_config.json> <path/to/train/slp> --val_labels <path/to/validation/slp>
```

Let me know if you run into any trouble.

Thanks,
Liezl

**Reply by ozgurcemozgen06** [reply]
Hi @roomrys  

Thanks a lot for your support. The only thing I want to ask to your answer is, we also need to include the multi_instance.json file(which contains the training specifications) to the train command, right? 

Best regards,
Özgür

**Reply by roomrys** [reply]
Hi @ozgurcemozgen06,

Yes! My mistake, I originally left out the `training_job_path` but have now included that argument in the psuedo-command above.

Thanks for pointing that out!
Liezl
